conftest.py

```diff
@@ -72,38 +85,39 @@ def test_table(request, sql_server):
         is_tmp_table = request.param.get("is_temp", True)
         table_options = request.param.get("param", {})
     sql_name, hook = sql_server
+    database = get_database_name(hook)
 
-    if isinstance(hook, SnowflakeHook):
+    if database == Database.SNOWFLAKE:
         params = {
             "conn_id": hook.snowflake_conn_id,
             "database": hook.database,
             "warehouse": hook.warehouse,
+            "schema": hook.schema,
         }
-    elif isinstance(hook, PostgresHook):
+    elif database == Database.POSTGRES:
         params = {"conn_id": hook.postgres_conn_id, "database": hook.schema}
-    elif isinstance(hook, SqliteHook):
+    elif database == Database.SQLITE:
         params = {"conn_id": hook.sqlite_conn_id, "database": "sqlite"}
-    elif isinstance(hook, BigQueryHook):
-        params = {
-            "conn_id": hook.gcp_conn_id,
-        }
+    elif database == Database.BIGQUERY:
+        params = {"conn_id": hook.gcp_conn_id, "schema": SCHEMA}
     else:
         raise ValueError("Unsupported Database")
-
     params.update(table_options)
     table = TempTable(**params) if is_tmp_table else Table(**params)
 
     yield table
 
-    if isinstance(hook, SqliteHook):
-        hook.run(f"DROP TABLE IF EXISTS {table.table_name}")
-        hook.run(f"DROP INDEX IF EXISTS unique_index")
-    elif isinstance(hook, SnowflakeHook):
-        hook.run(f"DROP TABLE IF EXISTS {table.table_name}")
-    elif not isinstance(hook, BigQueryHook):
-        # There are some tests (e.g. test_agnostic_merge.py) which create stuff which are not being deleted
-        # Example: tables which are not fixtures and constraints. This is an aggressive approach towards tearing down:
-        hook.run(f"DROP SCHEMA IF EXISTS {table.schema} CASCADE;")
+    if database == Database.BIGQUERY:
+        hook.run(f"DROP TABLE {table.qualified_name()}")
```

Comment:
Before, we were not dropping BQ tables :see_no_evil: I cleared up our BQ tables today in GCP